test/e2e/locators/recordset.ts

```diff
@@ -161,6 +165,68 @@ export default class RecordsetLocators {
     return container.locator('.side-panel-resizable');
   }
 
+  static getFacetById(container: Page | Locator, idx: number): Locator {
+    return container.locator(`.fc-${idx}`);
+  }
+
+  static getFacetHeaderButtonById(facet: Locator, idx: number): Locator {
+    return facet.locator(`.fc-heading-${idx} button`);
+  }
+
+  // get child of accordion group, sibling to accordion heading
+  static getFacetCollapse(facet: Locator): Locator {
+    return facet.locator('.accordion-collapse');
+  }
+
+  static getFacetOptions(facet: Locator): Locator {
+    return facet.locator('.chaise-checkbox label')
+  }
+
+  static getFacetOption(facet: Locator, optionIdx: number) {
+    return facet.locator(`.checkbox-${optionIdx}`);
+}
 
+  /* range facet selectors */
+  // there's integer/float/date/timestamp inputs
+  static getRangeInput(facet: Locator, className: string): Locator {
+    return facet.locator(`.${className}`);
+  }
+
+  static getInputClear(facet: Locator, className: string): Locator {
+    // NOTE: same as getRangeInput selector
+    return facet.locator(`.${className}`);
+  }
+
+  static getRangeSubmit(facet: Locator): Locator {
```

Comment:
You're adding functions to the "Recordset" class. So try to be more specific in the names. Just adding Facet should be enough (`getFacetRangeSubmit`, `getFacetHistogram`).

Reply:
By the way the `.range-input-submit-btn` could technically be returned by `getFacetRangeInputs ` that I suggested above. It's up to you if you would like to have it as a separate function or not.